client/src/components/convo/Convo.js

```diff
@@ -1,13 +1,95 @@
-import React from "react";
+import React, { useState, useEffect, useCallback } from "react";
+import DisplayAvatar from "../DisplayAvatar";
+
 import ConvoHeader from "./children/ConvoHeader";
 import SendMessage from "./children/SendMessage";
 
-const Conversation = () => {
+const Conversation = ({ currConvo }) => {
+	// Stores messages
+	const [currConvoMessages, setCurrConvoMessages] = useState({});
+
+	// Will scroll us to the bottom of a conversation
+	const setRef = useCallback((node) => {
+		if (node) node.scrollIntoView({ smooth: true });
+	}, []);
+
+	// Retrieves the conversation between sender and recipient
+	useEffect(() => {
+		// If it exists...
+		if (currConvo.conversationID) {
+			const retrieveMessages = async () => {
+				// Fetching user data
+				let data = await fetch(
+					`/api/conversations/${currConvo.conversationID}`
+				);
+				data = await data.json();
+				setCurrConvoMessages(data);
+			};
+			retrieveMessages();
+		}
+	}, [currConvo]);
+
+	const messagesCard = Object.keys(currConvoMessages).map((key, idx) => {
+		let {
+			conversationID,
+			sender,
+			content,
+			timeCreated,
+			_id
+		} = currConvoMessages[key];
+
+		// Keeps track of last message to auto scroll
+		let lastMessage = currConvoMessages.length - 1 === idx;
+
+		// Making our timeCreated variable more understandable
+		// let date = new Date(timeCreated).toLocaleDateString();
+		let time = new Date(timeCreated).toTimeString();
+
+		// Converting the time into 24 hour time
+		time = time.split(":");
+		time.pop();
+		time = time.join(":");
+
+		// Fixes issue of momentary gap in time where conversation ids don't match
+		if (currConvo.conversationID !== conversationID) return null;
```

Comment:
you can probably move this logic up so that it just returns null instead of calculating all the time above